lib/src/rules/unawaited_futures.dart

```diff
@@ -81,7 +82,11 @@ class _Visitor extends SimpleAstVisitor<void> {
     if (expr is AssignmentExpression) return;
 
     var type = expr?.staticType;
-    if (type?.isDartAsyncFuture == true) {
+    if (type == null) {
+      return;
+    }
+    if (type.isDartAsyncFuture ||
```

Comment:
It looks like this test will be duplicated when `implementsInterface` is invoked, so we can probably remove the first condition.

Reply:
Nice.  Thanks!